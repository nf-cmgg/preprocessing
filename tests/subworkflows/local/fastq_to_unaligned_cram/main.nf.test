nextflow_workflow {

    name "Test Workflow FASTQ_TO_UCRAM"
    script "subworkflows/local/fastq_to_unaligned_cram/main.nf"
    workflow "FASTQ_TO_UCRAM"

    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/local/fastq_to_unaligned_cram"

    test("fastq to unaligned cram") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', samplename:'test', single_end:false ], // meta map
                    [
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R1.fastq.gz", checkIfExists: true),
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R2.fastq.gz", checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                file(workflow.out.cram[0][1]).name,
                workflow.out.versions
            ).match()
        }

    }

}
