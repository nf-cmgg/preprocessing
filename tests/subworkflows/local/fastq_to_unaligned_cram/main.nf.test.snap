{
    "fastq to unaligned cram": {
        "content": [
            "test.unaligned.cram",
            [
                "versions.yml:md5,7bc1dacb3ee01e7e03a94cbbd644181c",
                "versions.yml:md5,a172a8ee6f026a436809819ed50e4e10"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T14:39:21.533097"
    }
}